nncf/quantization/algorithms/weight_compression/scale_estimation.py

```diff
@@ -246,18 +257,19 @@ def calculate_quantization_params(
         key = (config.mode, config.num_bits) + q_weights.shape + scale.shape
         if zp is not None:
             key += zp_shape
-        if key in ScaleEstimation.compress_decompress_cache:
-            compress_decompress_model = ScaleEstimation.compress_decompress_cache[key]["compress_decompress_model"]
-            compress_model = ScaleEstimation.compress_decompress_cache[key]["compress_model"]
-        else:
-            compress_decompress_model = backend_entity.get_compress_decompress_pipeline(
-                config, q_weights.shape, scale.shape, zp_shape
-            )
-            compress_model = backend_entity.get_compress_pipeline(config, q_weights.shape, scale.shape, zp_shape)
-            ScaleEstimation.compress_decompress_cache[key] = {
-                "compress_decompress_model": compress_decompress_model,
-                "compress_model": compress_model,
-            }
+        if config.mode != CompressWeightsMode.NF4:
```

Comment:
Is there a way to speed up compression weights to nf4 and decompression them with using OpenVINO?

Reply:
Could you implement conversion using gather from OpenVINO?